From b35185b8c991a0267840d2782aef4fcf19d6661b Mon Sep 17 00:00:00 2001
From: Olivier BIDEAU <olivier.bideau@st.com>
Date: Tue, 14 Nov 2023 14:31:33 +0100
Subject: [PATCH 427/841] feat(fdts): enable sdmmc2 for eMMC on STM32MP257F-DK

eMMC is mapped on sdmmc2.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 207903986a3dbab9c21f1b8aa164517247fd0f76 feat(fdts): enable sdmmc2 for eMMC on STM32MP257F-DK]
Signed-off-by: Olivier BIDEAU <olivier.bideau@st.com>
Change-Id: I4bb3fda332c4c530e7091f3f3ae208f97673a909
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/340213
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
---
 fdts/stm32mp257f-dk.dts | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/fdts/stm32mp257f-dk.dts b/fdts/stm32mp257f-dk.dts
index f47404db9..92e7c439e 100644
--- a/fdts/stm32mp257f-dk.dts
+++ b/fdts/stm32mp257f-dk.dts
@@ -195,6 +195,19 @@
 	status = "okay";
 };
 
+&sdmmc2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&sdmmc2_b4_pins_a &sdmmc2_d47_pins_a>;
+	non-removable;
+	no-sd;
+	no-sdio;
+	st,neg-edge;
+	bus-width = <8>;
+	vmmc-supply = <&vdd_emmc>;
+	vqmmc-supply = <&vddio2>;
+	status = "okay";
+};
+
 &usart2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&usart2_pins_a>;
-- 
2.47.2

